kernel/signal.c

```diff
@@ -756,9 +756,17 @@ long myst_signal_clone(
     return ret;
 }
 
-long myst_handle_host_signal(siginfo_t* siginfo, mcontext_t* mcontext)
+void myst_handle_host_signal(siginfo_t* siginfo, mcontext_t* mcontext)
 {
-    return _handle_one_signal(siginfo->si_signo, siginfo, mcontext);
+    _handle_one_signal(siginfo->si_signo, siginfo, mcontext);
+
+    /* The signal handler in the programming language runtime or the application
+     * might not return. If execution returns here, set the CPU context as the
+     * returned mcontext to continue execution. */
+    myst_sigreturn(mcontext);
```

Comment:
What would be an example for a handler that does not return?

Reply:
Would the FS/GS registers be correctly setup at this point?

Reply:
In .NET runtime, the signal handler could use IRETQ to resume execution.

Reply:
Is the signal handler invoked by `_handle_one_signal`? (The name seems to indicate that would be the case).
IN Mystikos, iretq is emulated. Therefore, should this code instead be in the emulation of that instruction?

Reply:
The follow is
enclave exception handler -> myst_handle_host_signal (signal.c) -> _handle_one_signal (signal.c)  -> _handler_wrapper (signal.c) -> user handler code (if registered)

We have correct iretq emulation already. The context restoring code is owned by .NET  (see https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/coreclr/pal/src/arch/amd64/context2.S#L154) if it decides to do early return.